evp.go

```diff
@@ -513,7 +510,34 @@ func newEvpFromParams(id C.int, selection C.int, params C.GO_OSSL_PARAM_PTR) (C.
 	}
 	var pkey C.GO_EVP_PKEY_PTR
 	if C.go_openssl_EVP_PKEY_fromdata(ctx, &pkey, selection, params) != 1 {
+		if vMinor < 2 {
```

Comment:
Include major? Admittedly, mainly for clarity--seems unlikely to be a significant stumbling point if OpenSSL 4 comes along. 😄 

```suggestion
		if vMajor == 3 && vMinor < 2 {
```

You have to explore a fair amount of context or know about this API in particular to know that the func will only be called when OpenSSL 3 is being used. Other funcs in this file do get called when using OpenSSL 1, so it seems reasonable to clarify in some way. `checkMajorVersion` would be another way, but has its cost.